### PR TITLE
Fix jsoc keyword not escaping a string for the query

### PR DESCRIPTION
**Pull request by nabobalis**
Supersedes: https://github.com/sunpy/sunpy/pull/7440/files

Bugfix for 5.0
Bugfix for 5.1
I would consider this a minor change.

### CODE REVIEW

File: sunpy/net/jsoc/jsoc.py
Comment (wtbarnes) [inline sunpy/net/jsoc/jsoc.py:653]:
```suggestion
            wavelength = f'{int(np.ceil(wavelength.to_value(u.AA)))}'
```

File: sunpy/net/jsoc/jsoc.py
Comment (wtbarnes) [inline sunpy/net/jsoc/jsoc.py:692]: This is the bugfix that is the actual focus of this PR right?

Comment (wtbarnes) [reply sunpy/net/jsoc/jsoc.py:692]: I'm also confused by this line. If as the comment says you are checking if this is a string or not, why not just do `if isinstance(value['value], str)`?

File: sunpy/net/jsoc/tests/test_jsoc.py
Comment (wtbarnes) [inline sunpy/net/jsoc/tests/test_jsoc.py:378]:
```suggestion
    )
```

Comment (wtbarnes) [reply sunpy/net/jsoc/tests/test_jsoc.py:378]: Am I going crazy? Shouldn't that that last `)` be indented at the same level as `responses`?

**Review by wtbarnes — APPROVED** [review]
This should really get a second review from someone who understands the JSOC client much better than I do.

File: sunpy/net/jsoc/tests/test_jsoc.py
Comment (nabobalis) [reply sunpy/net/jsoc/tests/test_jsoc.py:378]: No you are correct.

File: sunpy/net/jsoc/jsoc.py
Comment (nabobalis) [reply sunpy/net/jsoc/jsoc.py:692]: Err, I forget now

File: sunpy/net/jsoc/jsoc.py
Comment (Cadair) [inline sunpy/net/jsoc/jsoc.py:340]: If we care about these being sorted, why not just use a list? Taking a set and then converting it to a generator seems insane.

File: sunpy/net/jsoc/jsoc.py
Comment (Cadair) [inline sunpy/net/jsoc/jsoc.py:653]: As valid as these logic refactors are doing them in the middle of this PR which is fixing a legitimate bug, in code that most of us are not familiar with makes reviewing this PR *really* hard.

Comment (nabobalis) [reply sunpy/net/jsoc/jsoc.py:653]: Don't worry about it.

Comment (Cadair) [reply sunpy/net/jsoc/jsoc.py:653]: Well if you care about faster reviews then it might be something to think about :wink:

Comment (nabobalis) [reply sunpy/net/jsoc/jsoc.py:653]: *I won't reply to that with my actual thoughts.*

But yes, that is very true. I will endeavour to make sure my PR's are smaller in the future.

File: sunpy/net/jsoc/jsoc.py
Comment (nabobalis) [reply sunpy/net/jsoc/jsoc.py:340]: Fixed.

File: sunpy/net/jsoc/jsoc.py
Comment (nabobalis) [reply sunpy/net/jsoc/jsoc.py]: None of the unit tests failed for me locally when I did that, so I have no idea why I did that in the first place or the worrying one is that the unit tests don't cover this and I have broken something.

Only time and God can tell us.

**Comment by nabobalis** [comment]
Online fail is:

FAILED ../../.tox/py312-online/lib/python3.12/site-packages/sunpy/map/mapsequence.py::sunpy.map.mapsequence.MapSequence.save

Not due to this PR hopefully.

**Pull request merged**
